http_service/tests/test_readthrough_cache.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime, timedelta
+from unittest.mock import patch
+
+from bugbug_http.readthrough_cache import ReadthroughTTLCache
+
+
+class MockDatetime:
+    def __init__(self, mock_now):
+        self.mock_now = mock_now
+
+    def now(self):
+        return self.mock_now
+
+    def set_now(self, new_mock_now):
+        self.mock_now = new_mock_now
+
+
+class MockSleep:
+    def __init__(self, mock_datetime):
+        self.mock_datetime = mock_datetime
+        self.wakeups_count = 0
+
+        # count variable used to ensure that context switch into
+        # thread claling sleep occurred
+
+    def sleep(self, sleep_seconds):
+        self.wakeup_time = self.mock_datetime.now() + timedelta(seconds=sleep_seconds)
+        while self.mock_datetime.now() < self.wakeup_time:
+            pass
```

Comment:
If you use `time.sleep(0)`, you will yield control to the main thread, this might help avoiding hanging during the test